lib/streams/stream.ts

```diff
@@ -26,26 +26,32 @@ export default class Stream implements StreamInterface {
     return this._process.call(this, value, item);
   }
   private handleQueue(value: any, item: QueueElementInterface) {
-    this.targets.forEach((target: any) => {
+    for (let i = 0; i < this.targets.length; i++) {
+      let target = this.targets[i];
       let result = this.process(value, item);
       if (result) {
         target.write(result);
-      }
-    });
+      };
+    }
   }
+  // This likely has JITing problems
```

Comment:
Yea, we should measure this. I don't know if this code gets hot and can take many branches during it's life cycle.